src/api/dirmonitor/polling.c

```diff
@@ -0,0 +1,123 @@
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <SDL.h>
+
+enum {
+  EVENT_MODIFIED
+};
+
+struct file_info {
+  int fd;
+  struct stat file_stat;
+  int events;
+};
+
+struct dirmonitor_internal {
+  struct file_info* fds;
+  size_t fd_count;
+};
+
+  
+struct dirmonitor_internal* init_dirmonitor() {
+  struct dirmonitor_internal* monitor = calloc(1, sizeof(struct dirmonitor_internal));
+
+  monitor->fd_count = 0;
+  monitor->fds = NULL;
+  return monitor;
+}
+
+void deinit_dirmonitor(struct dirmonitor_internal* monitor) {
+  for (size_t i = 0; i < monitor->fd_count; ++i)
+    close(monitor->fds[i].fd);
+  free(monitor->fds);
```

Comment:
Is it not problematic with "removed" (`-1`) `fd`s?

Reply:
shouldn't be, `-1` is an invalid file descriptor and should make it error with `EBADF`